libenv/sysinfo.c

```diff
@@ -2158,8 +2163,10 @@ static int Linux_Suse_Version(EvalContext *ctx)
         }
         else
         {
-            sscanf(strversion, "VERSION = %s", strmajor);
-            sscanf(strpatch, "PATCHLEVEL = %s", strminor);
+            assert((sizeof(strmajor)) > 255); // TODO: static_assert()
+            assert((sizeof(strminor)) > 255); // TODO: static_assert()
+            sscanf(strversion, "VERSION = %255s", strmajor);
+            sscanf(strpatch, "PATCHLEVEL = %255s", strminor);
```

Comment:
There is an issue in these, though. They are defined as:

```
char strmajor[PRINTSIZE(major)], strminor[PRINTSIZE(minor)];
```

which means that the size is defined at runtime.

I suggest we change the declarations to be "big enough", following the way it is done everywhere else in the file. I just don't see why we want to optimize here to the byte while we are wasting so much memory in the others.

Reply:
We try to avoid hardcoded sizes, but I agree it should be constant, and a local macro for it might make sense.

Reply:
Fixed in my last push: I made these of size `CF_MAXVARSIZE`.